### ISSUE TITLE
Issues finding headers

### ISSUE THREAD

**Issue by chrisburr** [conda-forge/cling-feedstock#18]
The following was done inside a the official miniconda3 docker container that is based on Ubuntu and has almost nothing installed:
```
$ docker run --rm -it continuumio/miniconda3 bash
```

Create an environment:
```bash
$ conda create --name test-cling -c conda-forge/label/gcc7 -c conda-forge -c defaults python=3.7 cling
$ source activate test-cling
```

Cling then fails to find standard headers and segfaults when I try to compute `2+3`:

```bash
$ cling
In file included from input_line_1:1:
In file included from /opt/conda/envs/test-cling/bin/../lib/gcc/x86_64-conda_cos6-linux-gnu/7.3.0/../../../../x86_64-conda_cos6-linux-gnu/include/c++/7.3.0/new:39:
In file included from /opt/conda/envs/test-cling/bin/../lib/gcc/x86_64-conda_cos6-linux-gnu/7.3.0/../../../../x86_64-conda_cos6-linux-gnu/include/c++/7.3.0/x86_64-conda_cos6-linux-gnu/bits/c++config.h:533:
/opt/conda/envs/test-cling/bin/../lib/gcc/x86_64-conda_cos6-linux-gnu/7.3.0/../../../../x86_64-conda_cos6-linux-gnu/include/c++/7.3.0/x86_64-conda_cos6-linux-gnu/bits/os_defines.h:39:10: fatal error: 'features.h' file not found
#include <features.h>
         ^~~~~~~~~~~~

****************** CLING ******************
* Type C++ code and press enter to run it *
*             Type .q to exit             *
*******************************************
[cling]$ 2+3
Segmentation fault (core dumped)
```

If I manually set the include directories instead it seems okay:

```bash
$ cling -I $CONDA_PREFIX/x86_64-conda_cos6-linux-gnu/include/c++/7.3.0 -I $CONDA_PREFIX/x86_64-conda_cos6-linux-gnu/sysroot/usr/include

****************** CLING ******************
* Type C++ code and press enter to run it *
*             Type .q to exit             *
*******************************************
[cling]$ 2+3
(int) 5
[cling]$ #include <vector>
[cling]$ std::vector<double> a = {1, 2, 4}
(std::vector<double> &) { 1, 2, 4 }
```

**Comment by SylvainCorlay** [comment]
I can reproduce locally. I wonder if there is a build option to cling which we could populate with `$PREFIX`.

**Comment by chrisburr** [comment]
Looking at the ROOT recipe there are a couple of parts that might be relevant:
* [This patch from Gentoo's ROOT package is applied](https://github.com/conda-forge/staged-recipes/blob/72a0dff1135e234b9e1ae584f9dbe33c43e1e414/recipes/root/patches/root-6.12.06_cling-runtime-sysroot.patch)
* [Several options for setting the sysroot are used when running CMake](https://github.com/conda-forge/staged-recipes/blob/72a0dff1135e234b9e1ae584f9dbe33c43e1e414/recipes/root/build.sh#L16)

**Comment by SylvainCorlay** [comment]
`CLING_INCLUDE_PATHS`

sources:

https://github.com/root-project/cling/blob/d713ae033e49fae9d34f88607a254033b8c4e8c0/lib/Interpreter/CIFactory.cpp#L649-L653

cmake:

https://github.com/root-project/cling/blob/ee374738072fab73d2b9896c968a000cf0a1bfb4/CMakeLists.txt#L447-L451

**Comment by SylvainCorlay** [comment]
We need to find a means to specify that variable for OS X, windows, and Linux :(

**Comment by SylvainCorlay** [comment]
That said, `cling_include_deflt` does not contain the said path :(

**Comment by SylvainCorlay** [comment]
cc @vgvassilev

**Comment by SylvainCorlay** [comment]
so, on `Linux `

```cpp
-D CLING_INCLUDE_PATHS=$PREFIX/x86_64-conda_cos6-linux-gnu/include/c++/7.3.0:$PREFIX/x86_64-conda_cos6-linux-gnu/sysroot/usr/include
```

**Comment by SylvainCorlay** [comment]
Ah, `x86_64-conda_cos6-linux-gnu` is stored in the `$HOST`  environment variable. Opening a PR.

**Comment by SylvainCorlay** [comment]
Note, with the current build on conda-forge, I am seeing:

``` 
In file included from input_line_12:1:
In file included from /srv/conda/include/xtensor/xarray.hpp:19:
In file included from /srv/conda/include/xtensor/xcontainer.hpp:22:
In file included from /srv/conda/include/xtensor/xmath.hpp:24:
In file included from /srv/conda/include/xtensor/xaccumulator.hpp:17:
In file included from /srv/conda/include/xtensor/xexpression.hpp:22:
In file included from /srv/conda/include/xtensor/xtensor_forward.hpp:27:
In file included from /srv/conda/include/xtl/xoptional_sequence.hpp:20:
In file included from /srv/conda/include/xtl/xdynamic_bitset.hpp:12:
In file included from /srv/conda/bin/../lib/gcc/x86_64-conda_cos6-linux-gnu/7.3.0/../../../../x86_64-conda_cos6-linux-gnu/include/c++/7.3.0/climits:42:
In file included from /srv/conda/x86_64-conda_cos6-linux-gnu/sysroot/usr/include/limits.h:125:
In file included from /srv/conda/lib/clang/5.0.0/include/limits.h:37:
/usr/include/limits.h:145:5: error: function-like macro '__GLIBC_USE' is not defined
#if __GLIBC_USE (IEC_60559_BFP_EXT)
```

**Comment by chrisburr** [comment]
This is also affecting `root`, see https://github.com/conda-forge/root-feedstock/issues/2 for more details. It will be fixed by https://github.com/conda-forge/clangdev-feedstock/pull/45.
